Add docs for development (#94)
kubernetes-sigs/kwok: docs/contributing/development.md

# Development

> This doc walks you through how to run `kwok` and `kwokctl` in the local.

Reviewer: ```suggestion
This doc provides details on how to build and run `kwok` and `kwokctl` locally.
```

## Building

### Building `kwok` & `kwokctl`

``` bash
make build
```

> On `./bin/$(go env GOOS)/$(go env GOARCH)` will contain the freshly build binary upon a successful build.

Reviewer: ```suggestion
On a successful build, the binaries will be located in `./bin/$(go env GOOS)/$(go env GOARCH)`
```

### Building `kwok` image

On Linux:
``` bash
BINARY=kwok make build
IMAGE_PREFIX=local make image
```

On Darwin:
``` bash
CONTAINER_PLATFORMS=linux/$(go env GOARCH)
BINARY_PLATFORMS=${CONTAINER_PLATFORMS} BINARY=kwok make cross-build
IMAGE_PLATFORMS=${CONTAINER_PLATFORMS} IMAGE_PREFIX=local make cross-image
```

> Image `local/kwok:${tag}` will contain in `docker images`

Reviewer: ```suggestion
The image will be tagged as `local/kwok:${tag}` and can be found in `docker images`
```

> ### Start a local cluster with `kwokctl` and `kwok` on Docker

Reviewer: ```suggestion
### Starting a local cluster with locally built `kwokctl` and `kwok` using Docker
```
